packages/scene-composer/src/augmentations/components/three-fiber/anchor/AnchorWidget.tsx

```diff
@@ -113,16 +106,57 @@ export function AsyncLoadedAnchorWidget({
   useEffect(() => {
     setParent(node.parentRef ? getObject3DFromSceneNodeRef(node.parentRef) : undefined);
   }, [node.parentRef]);
-
   // Evaluate visual state based on data binding
   const visualState = useMemo(() => {
-    getCustomIconColor(ruleResult, setOverrideCustomColor);
-    const ruleTargetInfo = getSceneResourceInfo(ruleResult as string);
+    const checkTargetExists = typeof ruleResult === 'object' && 'target' in ruleResult;
+    const checkTargetMetadataExists = typeof ruleResult === 'object' && 'targetMetadata' in ruleResult;
+    if (checkTargetMetadataExists) {
+      // check overrideCustomColor is defined but not same as rule targetMetadata color
+      if (overrideCustomColor && overrideCustomColor !== ruleResult.targetMetadata?.color) {
+        setOverrideCustomColor(ruleResult.targetMetadata?.color);
+      }
+      // check overrideCustomColor is undefined but targetMetadata color is defined
+      if (!overrideCustomColor && ruleResult.targetMetadata?.color) {
+        setOverrideCustomColor(ruleResult.targetMetadata?.color);
+      }
+
+      // check overrideCustomIcon is defined but not same as rule targetMetadata icon
+      if (
+        overrideCustomIcon &&
+        overrideCustomIcon.iconName !== ruleResult.targetMetadata?.iconName &&
+        overrideCustomIcon.prefix !== ruleResult.targetMetadata?.iconPrefix
+      ) {
+        setOverrideCustomIcon({
+          prefix: ruleResult.targetMetadata?.iconPrefix,
+          iconName: ruleResult.targetMetadata?.iconName,
+        } as IconLookup);
+      }
+
+      // check overrideCustomIcon is undefined but targetMetadata icon is defined
+      if (ruleResult.targetMetadata && ruleResult.targetMetadata.iconName && ruleResult.targetMetadata.iconPrefix) {
+        setOverrideCustomIcon({
+          prefix: ruleResult.targetMetadata?.iconPrefix,
+          iconName: ruleResult.targetMetadata?.iconName,
+        } as IconLookup);
+      }
+    }
+    const ruleTargetInfo = getSceneResourceInfo(
+      checkTargetExists ? (ruleResult.target as string) : (ruleResult as string),
+    );
     // Anchor widget only accepts icon, otherwise, default to Info icon
     return ruleTargetInfo.type === SceneResourceType.Icon ? ruleTargetInfo.value : defaultIcon;
```

Comment:
all of above logic should run only when `ruleTargetInfo.type === SceneResourceType.Icon` is true. otherwise return default icon

Reply:
yes. ad also filtered the icons in anchor component as well since we have support only for icon.